veps/sig-compute/NNNN-cold-reboot-guest/cold-reboot-guest.md

```diff
@@ -0,0 +1,210 @@
+# VEP #NNNN: Guest initiated cold reboot support
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue created, which links to VEP dir in [kubevirt/enhancements] (not the initial VEP PR)
+- [ ] (R) Target version is explicitly mentioned and approved
+- [ ] (R) Graduation criteria filled
+
+## Overview
+
+<!--
+Provide a brief overview of the topic)
+-->
+
+Kubevirt currenty does not detect when guest reboots internally. The qemu process abstracts that away, the VMI is not terminated and new VM settings are not applied post-reboot. Lets call this warm reboot.
+
+This enhancement adds a toggle to the VMI API that instructs qemu to not reboot silently. Instead it will terminate and the reboot will be handled by the kubevirt controllers according to the selected runStrategy. As a (wanted) side effect, the VMI will be recreated using the up-to-date settings from the VM.
```

Comment:
> Instead it will terminate and the reboot will be handled by the kubevirt controllers according to the selected runStrategy.

So IIUC this means that when the guest shuts off the VMI will exit, is that correct?
Then, if the runStrategy dictates (e.g. Always), a new VMI will be created?

Reply:
From what I understood that is the idea. It will allow to carry over changes in a VM by recreating the VMI.